src/main/java/Duke.java

```diff
@@ -1,10 +1,377 @@
+import exceptions.IllegalCommandException;
+import exceptions.IllegalListException;
+import exceptions.IllegalTaskException;
+import exceptions.IllegalTaskRedoException;
+import list.*;
+
+import java.util.ArrayList;
+import java.io.File;
+import java.util.Scanner;
+import java.io.FileWriter;
+
 public class Duke {
+
+
+    public static final String CHAT_BOT_NAME = "Arthur";
+    public static final String FILE_PATH = "duke.txt";
+    public static ArrayList<TaskList> tasks = new ArrayList<>();
+    public static duke dukeList = new duke(CHAT_BOT_NAME);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        loadFromFile();
+        dukeList.greet();
+        processCommands();
     }
+
+    public static void loadFromFile() {
+        try {
+            File fd = new File(FILE_PATH); // create a File for the given file path
+            Scanner s = new Scanner(fd); // create a Scanner using the File as the source
+            readDataFile(s);
+            printOldList();
+        } catch (java.io.IOException e) {
+            dukeList.printNoSavedFile();
+        }
+    }
+
+    private static void readDataFile(Scanner s) {
+        int i = 0;
+        dukeList.printLoading();
+        while (s.hasNext()) {
+            selectCommand(s.nextLine());
+            if (s.nextLine().equals("true")) {
+                TaskList t = tasks.get(i);
+                t.setDone();
+                tasks.set(i, t);
+            }
+            i++;
+        }
+    }
+
+    private static void printOldList() {
+        try {
+            printAllLists();
+        } catch (IllegalListException e) {
+            dukeList.printEmptyFile();
+        }
+    }
+
+    public static void saveToFile() {
+        boolean hasSaved = true;
+        do {
+            hasSaved = writeToFile(hasSaved);
+        } while (!hasSaved);
+    }
+
+    private static boolean writeToFile(boolean hasSaved) {
```

Comment:
Since this method returns a boolean, you'd probably want to name the method such that it sounds like a question eg. `hasWrittenToFile`

Reply:
Resolved!